static/app/utils/reactQuery.spec.tsx

```diff
@@ -0,0 +1,112 @@
+import {render, screen} from 'sentry-test/reactTestingLibrary';
+
+import {useQuery} from 'sentry/utils/reactQuery';
+import RequestError from 'sentry/utils/requestError/requestError';
+import * as useApi from 'sentry/utils/useApi';
+
+type ResponseData = {
+  value: number;
+};
+
+beforeEach(() => {
+  jest.restoreAllMocks();
+});
+
+describe('reactQuery', function () {
+  describe('useQuery', function () {
+    it('can do a simple fetch', async function () {
+      const mock = MockApiClient.addMockResponse({
+        url: '/some/test/path/',
+        body: {value: 5},
+      });
+
+      const TestComponent = () => {
```

Comment:
Can’t use react hook testing library here?

Reply:
IMO it's slightly better to test hooks within a component when you can, and in this case I like testing it this way because 1. it depends on the `QueryClientProvider` in the render function and 2. it's a nice example of how to use the hook. But if we want to be consistent with testing hooks I can change it, will just need to pass in a query client to the options I believe.